### DISCUSSION TITLE
Unity version is not supported issue

### DISCUSSION THREAD

**Discussion by TamNT96** [codemagic-ci-cd/codemagic-docs#2451] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Unity

### Steps to reproduce

1. Setup a workflow with environment unity version match with version in the document:
https://docs.codemagic.io/knowledge-others/install-unity-version/
2. Start the build.

### Expected results

 Prepairing build machine should complete with: 
== Use Unity version 2021.3.9f1 ==
Downloading and installing Unity from Codemagic storage... done in 2 min 1 second

### Actual results

== Use Unity version 2021.3.9f1 ==
Downloading and installing Unity from Codemagic storage...

Build failed :|
Required Unity version is not supported via `environment` configuration.
Please see https://docs.codemagic.io/knowledge-others/install-unity-version/ for the list of supported versions or how to install from Unity Hub.

### Build id (optional)

660a4a728846bd3ba9702f90

**Comment by himesh-cm** [comment]
Hi @TamNT96,
We are looking into this and will update the thread once I hear from the team.

**Comment by CristianCanalesAnd** [comment]
Hi, 

I have the same problem, I successfully build my app 18 days ago. 
Today I try do generate a new build and fail it with the same error.

Unity version: 
2023.1.17f1

Build id: 
660a8c3fa4d74f4335be47ec

Build failed :|
Required Unity version is not supported via `environment` configuration.
Please see https://docs.codemagic.io/knowledge-others/install-unity-version/ for the list of supported versions or how to install from Unity Hub.

**Comment by CristianCanalesAnd** [comment]
Hi,

I was able to build the app changing the version of xcode from 15.3 to 15.2

**Reply by mikhail-tokarev** [reply]
Hey, it should work with Xcode 15.3 as well. Can you give it another try?
